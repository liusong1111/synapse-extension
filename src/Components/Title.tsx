import * as React from "react";

interface AppProps {
  title: string
}

interface AppState {}

export default class extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  render() {
    return (
      <h3>
        {this.props.title}
      </h3>
    )
  }
}