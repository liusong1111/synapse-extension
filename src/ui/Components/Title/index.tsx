import React from 'react';

interface AppProps {
  title: string;
  testId: string;
}

interface AppState {}

export default class extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  render() {
    return <h3 data-testid={this.props.testId || 'title'}>{this.props.title}</h3>;
  }
}
