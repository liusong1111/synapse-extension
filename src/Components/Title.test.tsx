import * as React from 'react'
import { shallow } from 'enzyme'
import Title from './Title'

describe('Title', () => {
  let wrapper
  beforeEach(() => wrapper = shallow(<Title title={''} />))

  it('should render Title', () => {
    expect(wrapper.find('div').length).toEqual(1)
  });

  it('should render title text', () => {
    wrapper.setProps({ title: 'Import Mnemonic'})
    expect(wrapper.text()).toEqual('Import Mnemonic')
  });


})