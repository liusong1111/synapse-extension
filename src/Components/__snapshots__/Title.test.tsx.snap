// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly 1`] = `
<h3
  data-testid="test-title"
/>
`;
